graylog2-server/src/main/java/org/graylog2/decorators/DecoratorProcessor.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.decorators;
+
+import com.google.inject.ImplementedBy;
+import org.graylog2.indexer.results.ResultMessage;
+import org.graylog2.rest.resources.search.responses.SearchResponse;
+
+import java.util.List;
+import java.util.Optional;
+
+@ImplementedBy(DecoratorProcessorImpl.class)
+public interface DecoratorProcessor {
+    List<ResultMessage> decorate(List<ResultMessage> messages, Optional<String> stream);
+    SearchResponse decorate(SearchResponse searchResponse, Optional<String> stream);
```

Comment:
That is exactly the rationale behind this, ease of implementing a decorator vs. completeness. I also share your conclusion of using only the latter, but in the next iteration.